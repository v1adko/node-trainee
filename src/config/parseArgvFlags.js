function capitalizeFirstLetter(string) {
  return string.charAt(0).toUpperCase() + string.slice(1);
}

const flagToCamelCase = flag => flag
  .substr(2)
  .split('-')
  .map((p, index) => {
    if (index > 0) {
      return capitalizeFirstLetter(p);
    }
    return p;
  })
  .join('');

const parseArgv = () => {
  const stringFlags = process.argv.filter(arg => arg.indexOf('--') === 0);
  const flags = {};
  stringFlags.forEach((flag) => {
    flags[flagToCamelCase(flag)] = true;
  });
  return flags;
};

export default parseArgv;
