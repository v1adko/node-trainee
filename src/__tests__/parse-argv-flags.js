import parseArgvFlags from '../lib/parse-argv-flags';

describe('Test parse argv flags', () => {
  it('should create db connection', () => {
    process.argv = ['--testArg'];
    const result = parseArgvFlags();

    expect(result).toMatchSnapshot();
  });

  it('should create db connection', () => {
    process.argv = ['--test-arg'];
    const result = parseArgvFlags();

    expect(result).toMatchSnapshot();
  });

  it('should create db connection', () => {
    process.argv = [];
    const result = parseArgvFlags();

    expect(result).toEqual({});
  });

  it('should create db connection', () => {
    process.argv = ['some arg'];
    const result = parseArgvFlags();

    expect(result).toEqual({});
  });
});
