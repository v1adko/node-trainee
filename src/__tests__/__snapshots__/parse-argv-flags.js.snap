// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test parse argv flags should create db connection 1`] = `
Object {
  "testArg": true,
}
`;

exports[`Test parse argv flags should create db connection 2`] = `
Object {
  "testArg": true,
}
`;
